File: examples/runnable/01b-hello-temporal-anywhere/go/greeting.go
// Env-driven Hello Temporal (Lab 1.2b Docker / 1.2c Cloud), Go version.
package main

import (
	"context"
	"time"

	"go.temporal.io/sdk/workflow"
)

const TaskQueue = "hello-anywhere"

func ComposeGreeting(ctx context.Context, name string) (string, error) {
	return "Hello, " + name + " from a Temporal Activity", nil
}

func GreetingWorkflow(ctx workflow.Context, name string) (string, error) {
	ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
		StartToCloseTimeout: 10 * time.Second,
	})
	var greeting string
	err := workflow.ExecuteActivity(ctx, ComposeGreeting, name).Get(ctx, &greeting)
	return greeting, err
}

File: examples/runnable/01b-hello-temporal-anywhere/python/greeting.py
"""Workflow + Activity for the env-driven Hello lab (same as 01-hello-temporal)."""

from datetime import timedelta

from temporalio import activity, workflow

TASK_QUEUE = "hello-anywhere"


@activity.defn
async def compose_greeting(name: str) -> str:
    return f"Hello, {name} from a Temporal Activity"


@workflow.defn
class GreetingWorkflow:
    @workflow.run
    async def greet(self, name: str) -> str:
        return await workflow.execute_activity(
            compose_greeting, name, start_to_close_timeout=timedelta(seconds=10)
        )
